app/models/foreman_leapp/preupgrade_report.rb

```diff
@@ -0,0 +1,37 @@
+module ForemanLeapp
+  class PreupgradeReport < ::Report
+    has_many :preupgrade_report_entries
+    has_many :messages, :through => :preupgrade_report_entries
+
+    def self.create_report(host, date, status, data)
+      report = PreupgradeReport.create(:host => host,
+                                       :reported_at => date,
+                                       :status => status)
+      if data[:entries]
+        data[:entries].each do |entry|
+          msg = Message.find_or_create(entry[:title])
+          msg.attributes = {}
+          msg.save!
+          PreupgradeReportEntry.create!(:message => msg,
+                                        :severity => :info,
+                                        :preupgrade_report => report,
+                                        :hostname => host.hostname,
+                                        :title => entry[:title],
+                                        :actor => entry[:actor],
+                                        :summary => entry[:summary],
+                                        :audience => entry[:audience],
+                                        :severity => entry[:severity],
+                                        :tags => entry[:tags],
+                                        :details => entry[:details],
+                                        :leapp_run_id => data[:leapp_run_id])
+        end
+      end
+      return report
+    end
+
+
+    def self.find_report(report_id)
+      return PreupgradeReport.where(:id => report_id).first
+    end
```

Comment:
Is this necessary? Why create a custom `PreupgradeReport.find_report(report_id)` method when we can do just `PreupgradeReport.find(report_id)`?